modules/scripts/startup-script/files/early_run_hotfixes.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script applies fixes to VMs that must occur early in boot. For example,
+# when yum or apt repositories are misconfigured, preventing most package
+# operations from completing successfully.
+
+source /etc/os-release
+
+if [[ "$PRETTY_NAME" == "CentOS Linux 7 (Core)" ]]; then
+	echo "Applying hotfixes for CentOS 7"
+	if grep -q '^mirrorlist' /etc/yum.repos.d/CentOS-Base.repo; then
```

Comment:
I think this needs to be applied to other `*.repo` files as well.  I can see the following CentOS repos on image `centos-7-v20240611`:
```
CentOS-Base.repo
CentOS-CR.repo
CentOS-Debuginfo.repo
CentOS-fasttrack.repo
CentOS-Media.repo
CentOS-Sources.repo
CentOS-Vault.repo
CentOS-x86_64-kernel.repo
```

Reply:
Discussed in chat - the other repos do not seem to be enabled.

Reply:
The only configuration that contains an "enabled" is CentOS-Base.repo. The built-in configurations in CentOS-Vault.repo are (IMO) superior to grep/sed of CentOS-Base.repo to use vault.centos.org but harder to automate enabling. This is functionally equivalent to disabling the repos in CentOS-Base.repo and enabling the 7.9.2009 repos in CentOS-Vault.repo.